### PR TITLE
clang-format: Use BeforeColon instead of BeforeComma in examples

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #8154, #8153

### CODE REVIEW

**Comment by tamiko** [comment]
@tjhei I prefer `BeforeComma`, but this is how it would look like when changing to `BeforeColon`.

**Comment by tjhei** [comment]
Nice! Well, I guess this is a matter of preference, so maybe people want to vote on this comment? Please:
- :+1: if you like the comma at the end (this PR)
- :-1:  if you like it how it is
- :woman_shrugging: if you don't care

**Comment by masterleinad** [comment]
Note that the failing `Travis` check shows the remaining changes this pull request introduces.

File: examples/step-42/step-42.cc [hunk 1]
```diff
@@ -152,14 +152,11 @@ namespace Step42
...
-    : kappa(E / (3 * (1 - 2 * nu)))
-    , mu(E / (2 * (1 + nu)))
-    , sigma_0(sigma_0)
-    , gamma(gamma)
-    , stress_strain_tensor_kappa(kappa *
-                                 outer_product(unit_symmetric_tensor<dim>(),
-                                               unit_symmetric_tensor<dim>()))
-    , stress_strain_tensor_mu(
+    : kappa(E / (3 * (1 - 2 * nu))), mu(E / (2 * (1 + nu))), sigma_0(sigma_0),
+      gamma(gamma), stress_strain_tensor_kappa(
```
Comment (bangerth) [inline examples/step-42/step-42.cc]: I veto this monstrosity. clang-format is bad enough, but the previous version at least put each variable on its own line. Here, it starts the second variable on the same line just to break the initializer onto the next line. This makes no sense.

**Comment by tamiko** [comment]
Let's go with the majority decision.

**Pull request closed without merging**